RELEASE/scripts/sl_ascend.ash

```diff
@@ -15049,6 +15049,35 @@ void print_help_text()
 	print_html("");
 }
 
+void safe_preference_reset_wrapper(int level)
+{
+	if(level <= 0)
+	{
+		sl_begin();
+	}
+	else
+	{
+		boolean succeeded;
+		try
+		{
+			safe_preference_reset_wrapper(level-1);
+			succeeded = true;
+		}
+		finally
+		{
+			restoreAllSettings();
+			restoreSetting("autoSatisfyWithCoinmasters");
+			restoreSetting("autoSatisfyWithNPCs");
+			restoreSetting("removeMalignantEffects");
+			restoreSetting("kingLiberatedScript");
+			restoreSetting("afterAdventureScript");
+			restoreSetting("betweenAdventureScript");
+			restoreSetting("betweenBattleScript");
+			restoreSetting("counterScript");
```

Comment:
We should maybe investigate moving these manual `restoreSetting` calls to `restoreAllSettings`. There may be a reason we shouldn't, though. From the looks of it this is the only place these restores are called, while `restoreAllSettings` is called in some other places... Seems like a potential oversight?

Reply:
I think this is a subtle oversight by me, thanks for the catch! The other `restoreAllSettings` call sites are

* in `kingsool.ash` when we free the king. We probably want to restore these properties here, since the user is done using `sl_ascend`.
* in `sl_interruptCheck`, which triggers an `abort()` when `sl_interrupt` is set. We probably don't want to restore these properties here (reasoning follows)

One reason to not unset `afterAdventureScript` on a non-bedtime abort is if the user is interrupting a fight to finish it manually - in that case, we probably still want to run the automatic post-adventure script for buff maintenance and so on. For now I'll restore the old functionality of restoring these properties at bedtime, but restoring other properties on abort.

Reply:
Wait, never mind. `restoreAllSettings()` will definitely reset all of those settings (since they're present in `defaults.txt`), so the hard-coded list is redundant and never did anything. Weeeeird.